src/modules/rtpengine/rtpengine.c

```diff
@@ -3441,7 +3493,17 @@ static int parse_flags(struct ng_flags_parse *ng_flags, struct sip_msg *msg,
 					ng_flags->transport = 0x103;
 				else if(str_eq(&key, "direction"))
 					bencode_list_add_str(ng_flags->direction, &val);
-				else
+				else if(str_eq(&key, "from-tags")) {
+					err = "missing value";
+					if(!val.s)
+						goto error;
+
+					if(!ng_flags->from_tags) {
+						ng_flags->from_tags =
+								bencode_list(ng_flags->dict->buffer);
+					}
+					bencode_list_add_str(ng_flags->from_tags, &val);
+				} else
```

Comment:
Ok if this is how you want to do it, but some suggestions for alternatives: `subscribe request` can take either a single `from-tag` or multiple `from-tags` as a list, and would treat them the same. So here you could also treat "from-tag" and "from-tags" the same, and making building the list conditional on the opmode.

Or: omit handling altogether and require a modern version of rtpengine which parses the flags internally (`parsed_by_module` false). In this case you can use syntax like `rtpengine_blah("flag flag ... from-tags=[tag1 tag2 tag3] ...")`

Up to you, just a hint.